.github/workflows/testing.yml

```diff
@@ -9,116 +9,69 @@ on:
-    defaults:
-      run:
-        shell: bash
+      matrix:
+        os: [windows-latest, ubuntu-latest, macos-latest]
+        environment: [
+            ci-py309-min-deps,
+            ci-py310-min-optional-deps,
+            ci-py312-latest-deps,
+            ci-py312-latest-optional-deps
+        ]
+    runs-on: ${{ matrix.os }}
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-python@v4
+      - uses: actions/checkout@v4
+      - uses: prefix-dev/setup-pixi@v0.8.1
         with:
-            python-version: ${{ matrix.python-version }}
-        name: 'Setup python'
-      - shell: bash {0}
-        run: ./build_tools/github/install.sh
-        name: 'Install skrub'
-        env:
-          DEPS_VERSION: ${{ matrix.dependencies-version }}
-      - shell: bash {0}
-        run: |
-          cp $GITHUB_WORKSPACE/pyproject.toml .
-          $GITHUB_WORKSPACE/build_tools/github/test.sh
-        working-directory: ${{ runner.temp }}
-        name: 'Run tests'
-      - uses: codecov/codecov-action@v3
+          pixi-version: v0.24.0
+          environments: ${{ matrix.environment }}
+          # we can freeze the environment and manually bump the dependencies to the
+          # latest version time to time.
+          frozen: true
+
+      - name: Run tests
+        run: pixi run -e ${{ matrix.environment }} test -n 3
+
+      - name: Upload coverage reports to Codecov
+        uses: codecov/codecov-action@v4.4.1
         with:
-          directory: ${{ runner.temp }}
-          verbose: true
-        if: success()
-        name: 'Upload coverage to CodeCov'
+          token: ${{ secrets.CODECOV_TOKEN }}
+          slug: skrub-data/skrub
 
-  run_nightly:
+  check_run_nightly:
     runs-on: ubuntu-latest
     outputs:
       run: ${{ github.event_name == 'schedule' || contains(steps.get_head_commit_message.outputs.COMMIT_MSG, '[deps nightly]') }}
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
         with:
           ref: ${{ github.event.pull_request.head.sha }}
       - id: get_head_commit_message
         name: get head commit message
         run: echo "COMMIT_MSG=$(git show -s --format=%s)" >> "$GITHUB_OUTPUT"
 
-  nightly:
-    needs: run_nightly
-    if: ${{ needs.run_nightly.outputs.run == 'true' }}
+  test_against_nightly:
+    needs: check_run_nightly
+    if: ${{ needs.check_run_nightly.outputs.run == 'true' }}
     runs-on: ubuntu-latest
     name: test against nighlty dependencies
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - uses: actions/setup-python@v4
         with:
             python-version: "3.11"
         name: 'Setup python'
       - shell: bash {0}
-        run: ./build_tools/github/install.sh
-        name: 'Install skrub'
-        env:
-          INSTALL_NIGHTLY: "true"
-          DEPS_VERSION: "dev"
+        run: |
```

Comment:
I guess there's no way to specify this environment with pixi?

Reply:
Here, it does not make sense to use `pixi` because you don't want to pin the nightly uild.

In this manner, we also test that our `pip` command of `pyproject.toml` work